packages/host/app/components/operator-mode/card-url-bar.gts

```diff
@@ -0,0 +1,190 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { BoxelInput } from '@cardstack/boxel-ui';
+import { svgJar } from '@cardstack/boxel-ui/helpers/svg-jar';
+import { action } from '@ember/object';
+import { service } from '@ember/service';
+import { trackedFunction } from 'ember-resources/util/function';
+import type { CardDef } from 'https://cardstack.com/base/card-api';
+import type CardService from '../../services/card-service';
+import { cardTypeDisplayName } from '@cardstack/runtime-common';
+import { or, not } from '@cardstack/boxel-ui/helpers/truth-helpers';
+
+interface Signature {
+  Element: HTMLElement;
+  Args: {
+    url: URL;
+    onURLChange: (url: URL) => void;
+    card: CardDef | null;
+    isInvalid: boolean;
+    errorMessage: string | null;
+  };
+}
+
+export default class CardURLBar extends Component<Signature> {
+  <template>
+    <div
+      id='card-url-bar'
+      class={{this.cssStyle}}
+      data-test-card-url-bar
+      ...attributes
+    >
+      {{#if (not this.isFocused)}}
+        <div class='realm-info' data-test-card-url-bar-realm-info>
+          <img src={{this.realmIcon}} />
+          <span>in
+            {{if this.realmName this.realmName 'Unknown Workspace'}}</span>
+        </div>
+      {{/if}}
+      <div class='input'>
+        {{svgJar 'icon-globe' width='22px' height='22px'}}
+        <BoxelInput
+          class='url-input'
+          @value={{if
+            (or this.isFocused (not this.cardDisplayName))
+            this.url
+            this.cardDisplayName
+          }}
+          @onInput={{this.onInput}}
+          @onKeyPress={{this.onKeyPress}}
+          @onFocus={{this.toggleFocus}}
+          @onBlur={{this.toggleFocus}}
+          data-test-card-url-bar-input
+        />
+      </div>
+      {{#if this.showErrorMessage}}
+        <div class='error-message' data-test-card-url-bar-error>
+          <span>{{if this.isInvalid this.errorMessage @errorMessage}}</span>
```

Comment:
This whole block reads confusingly. You have logic in `showErrorMessage` that operates on `this.errorMessage` and `@errorMessage`, and then there is also logic in the template here that again operates on the same params. I think this can be simplified